crates/astria-conductor/src/celestia/reporting.rs

```diff
@@ -0,0 +1,87 @@
+//! Various newtype-wrappers to emit serde-serialized tracing event fields.
+use serde::ser::{
+    Serialize,
+    SerializeSeq,
+    SerializeStruct,
+};
+use telemetry::display::hex;
+
+use super::{
+    ReconstructedBlock,
+    ReconstructedBlocks,
+};
+use crate::block_cache::GetSequencerHeight;
+
+pub(super) struct ReportSequencerHeights<'a, T>(pub(super) &'a [T]);
+
+impl<'a, T> Serialize for ReportSequencerHeights<'a, T>
+where
+    T: GetSequencerHeight,
+{
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut seq = serializer.serialize_seq(Some(self.0.len()))?;
+        for elem in self.0 {
+            seq.serialize_element(&elem.get_height().value())?;
+        }
+        seq.end()
+    }
+}
+
+pub(super) struct ReportReconstructedBlocks<'a>(pub(super) &'a ReconstructedBlocks);
+impl<'a> Serialize for ReportReconstructedBlocks<'a> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        const FIELDS: [&str; 4] = [
+            "celestia_height",
+            "sequencer_namespace",
+            "rollup_namespace",
+            "reconstructed_blocks",
+        ];
+        let mut state = serializer.serialize_struct("ReconstructedBlocksInfo", FIELDS.len())?;
+        state.serialize_field(FIELDS[0], &self.0.celestia_height)?;
+        state.serialize_field(FIELDS[1], &hex(&self.0.sequencer_namespace.as_bytes()))?;
```

Comment:
namespaces are hex-formatted bytes here (as opposed to base64). This seems to be in-line with block explorers like celenium.